Add test for RTCIceCandidatePair interface

dontcallmedom opened the pull request:
see https://github.com/w3c/webrtc-pc/issues/2978

dontcallmedom committed:
Add test for RTCIceCandidatePair interface

see https://github.com/w3c/webrtc-pc/issues/2978

alvestrand commented on webrtc/idlharness.https.window.js:
```diff
@@ -65,13 +65,18 @@ function asyncInitTransports() {
     idlTestObjects.iceTransport = iceTransport;
+
+    const iceCandidatePair = iceTransport.getSelectedCandidatePair();
+    assert_true(iceCandidatePair instanceof RTCIceCandidatePair,
```
For development, you should test that you get here and get a candidate pair - assert_not_null(iceCandidatePair).
 think the test will only work if the candidate is selected before you get this far - which is timing dependent, but can't be depended on without waiting on a state change.

dontcallmedom replied:
indeed; amended the test to trigger and await the proper ice connection state

dontcallmedom committed:
Fix test to wait for ICE connection

per https://github.com/web-platform-tests/wpt/pull/46655#pullrequestreview-2104336445

dontcallmedom committed:
Fix assert name

dontcallmedom commented:
(this should probably wait for #46647 to be merged)
